src/borg/remote.py

```diff
@@ -171,6 +276,27 @@ def open(self, path, create=False, lock_wait=None, lock=True, exclusive=None, ap
         self.repository.__enter__()  # clean exit handled by serve() method
         return self.repository.id
 
+    def inject_exception(self, kind):
```

Comment:
I wonder if we want to be able to test this against older borg versions, it‘s hard to unit test this completely. If not i can have a look how to move this into a subclass in the testsuite.